src/state/metadata/reducer.ts

```diff
@@ -33,7 +35,8 @@ import {
     FileInfo,
 } from "./types";
 
-export const initialState = {
+export const initialState: MetadataStateBranch = {
+    imageDataset: RequestImageDatasetToUse(),
```

Comment:
I don't think this belongs here. I would create a reducer/actions/logics in `image-dataset`. This branch of state is specifically metadata ABOUT the dataset, so also storing the dataset here seems strange